### PR TITLE
Adds React Native 0.40 support to Ignite 2.

### PR DESCRIPTION
**Pull request by skellock**
Also, when we run `ignite new MyApp`, we will default to installing the `ignite-*` plugins from our hard drive instead of `npm`.

This is pretty much what we always want to do (for now).  Once we get closer to launch, we can remove this.

`ignite new MyApp --live` to bypass if you really need to.

### CODE REVIEW

File: ignite-base/App/Containers/LoginScreen.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ class LoginScreen extends React.Component {
     const editable = !fetching
     const textInputStyle = editable ? Styles.textInput : Styles.textInputReadonly
     return (
-      <ScrollView contentContainerStyle={{justifyContent: 'center'}} style={[Styles.container, {height: this.state.visibleHeight}]} keyboardShouldPersistTaps>
+      <ScrollView contentContainerStyle={{justifyContent: 'center'}} style={[Styles.container, {height: this.state.visibleHeight}]} keyboardShouldPersistTaps='always'>
```
Comment (kevinvangelder) [inline ignite-base/App/Containers/LoginScreen.js]: Did you intend to update this non-2.0 file?

Comment (skellock) [inline ignite-base/App/Containers/LoginScreen.js]: No! Gah.  Great save.  Will change in the morning.

File: packages/ignite-basic-structure/templates/Unholy/Containers/LoginScreen.js [hunk 2]
```diff
@@ -112,7 +112,7 @@ class LoginScreen extends React.Component {
     const editable = !fetching
     const textInputStyle = editable ? Styles.textInput : Styles.textInputReadonly
     return (
-      <ScrollView contentContainerStyle={{justifyContent: 'center'}} style={[Styles.container, {height: this.state.visibleHeight}]} keyboardShouldPersistTaps>
+      <ScrollView contentContainerStyle={{justifyContent: 'center'}} style={[Styles.container, {height: this.state.visibleHeight}]} keyboardShouldPersistTaps='always'>
```
Comment (GantMan) [inline packages/ignite-basic-structure/templates/Unholy/Containers/LoginScreen.js]: Is this something new in 0.40?  BC, I've been teasing a short blog on the crazy prop for a while now, and if this is new, then blog++;

File: packages/ignite-basic-structure/templates/package.json.unholy.ejs [hunk 3]
```diff
@@ -23,46 +23,46 @@
...
     "redux": "^3.6.0",
     "redux-logger": "^2.7.4",
-    "redux-persist": "^3.5.0",
-    "redux-saga": "^0.13.0",
-    "reduxsauce": "0.2.0",
-    "seamless-immutable": "^6.3.0"
+    "redux-persist": "^4.1.0",
+    "redux-saga": "^0.14.3",
+    "reduxsauce": "0.4.1",
+    "seamless-immutable": "^7.0.1"
```
Comment (GantMan) [inline packages/ignite-basic-structure/templates/package.json.unholy.ejs]: I love that you updated all of these.  I know some were necessary, but it's always a key update.

File: packages/ignite/src/plugins/default/commands/new.js [hunk 4]
```diff
@@ -29,35 +29,51 @@ module.exports = async function (context) {
...
+  const igniteDevPackagePrefix = parameters.options.live || `${__dirname}/../../../../../ignite-`
+
   // First we ask!
   const answers = await context.prompt.ask(installWalkthrough)
   // then we kick off (TODO: Would be awesome to have this kick off during questions)
   // we need to lock the RN version here
   info('Creating new RN project')
   // TODO make sure `react-antive --version has react-native-cli 2.x otherwise failure`
-  await system.run(`react-native init ${projectName} --version 0.38.0`)
+  const reactNativeVersion = '0.40.0'
```
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: Thoughts on storing this in package.json?  I have not idea who/what would also care about this, but centralizing versions feels right to me.  *shrug*

File: packages/ignite/src/plugins/default/commands/new.js [hunk 5]
```diff
@@ -29,35 +29,51 @@ module.exports = async function (context) {
...
   if (answers['vector-icons'] === 'react-native-vector-icons') {
-    await system.run('ignite add vector-icons')
+    await system.run(`ignite add ${igniteDevPackagePrefix}vector-icons`)
   }
 
   debug(answers)
 
-  info('cd to your new project')
+  info(`Please type cd ${projectName}`)
   info('YOU MUST NOW RUN `react-native link` Sorry we are not doing that for you yet')
```
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: you can kill this message now that line 61 performs the necessary evil.

File: packages/ignite/src/plugins/default/commands/new.js [hunk 6]
```diff
@@ -29,35 +29,51 @@ module.exports = async function (context) {
...
   }
 
+  // Install the ignite-* packages from the local dev directory.
+  //
+  // This is pretty much always what we while we dev.
+  //
+  // To test what live is like, you can run `ignite new FooTown --live`.
+  //
+  // TODO(steve): Don't forget to remove this when we launch... open to better ways of handling it.
+  const igniteDevPackagePrefix = parameters.options.live || `${__dirname}/../../../../../ignite-`
```
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: /gif me gusta

File: packages/ignite-basic-structure/templates/Unholy/Containers/LoginScreen.js (hunk 2 above)
Comment (skellock) [inline packages/ignite-basic-structure/templates/Unholy/Containers/LoginScreen.js]: Yes.  This is new.  And it's AWESOME.  There's a few different options now.

File: packages/ignite/src/plugins/default/commands/new.js (hunk 6 above)
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: What you had was right.  This is just a developer quality-of-life update.  Short term.

File: packages/ignite/src/plugins/default/commands/new.js (hunk 4 above)
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: Maybe default to "latest ignite supported one" with an optional override option?

```sh
ignite new HoldMyBeer --react-native 0.41.0-rc-0
```

File: packages/ignite/src/plugins/default/commands/new.js (hunk 6 above)
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: me gusta means I like it.  I know the meme is a silly face.   I'm saying I like what you did.

File: packages/ignite/src/plugins/default/commands/new.js (hunk 4 above)
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: OMG yes.  That sir, is perfect.

File: packages/ignite/src/plugins/default/commands/new.js (hunk 5 above)
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: For the record, you're going to hell for writing 61.

But yes, I'll remove this message.  And join you in hell for aiding & abetting.

🚔

File: packages/ignite/src/plugins/default/commands/new.js (hunk 6 above)
Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: Ha!  I always thought it was: my bad.  Don't know what I thought that.

Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: Apparently it's spanish too.  I thought it was latin.

Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: And now everyone knows how uneducated I really am.

Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: That's oke I dunt read good sometimes.  Welcome!

File: packages/ignite/src/plugins/default/commands/new.js (hunk 5 above)
Comment (GantMan) [inline packages/ignite/src/plugins/default/commands/new.js]: 😈

Comment (skellock) [inline packages/ignite/src/plugins/default/commands/new.js]: I updated the message to what we have in ignite 1.

**Pull request merged**